lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
-
-        # mask transits in design matrix
-        MX = M(X)
-
-        if use_gp:
-            # We use a Gaussian Process to model the long term trend.
-            # We do this by estimating the long term trend y by applying the
-            # preliminary PLD model defined above and subtracting it from the raw light curve.
-            # The "in transit" cadences are masked out in this step to prevent the
-            # long term approximation from over-fitting the transits.
-            XTX = np.dot(MX.T, MX)
-            XTX[np.diag_indices_from(XTX)] += 1e-8
-            XTy = np.dot(MX.T, M(rawflux))
-            y = M(rawflux) - np.dot(MX, np.linalg.solve(XTX, XTy))
-
-            # Estimate the amplitude parameter of a Matern-3/2 kernel GP
-            # by computing the standard deviation of y.
-            amp = np.nanstd(y)
-            tau = gp_timescale  # tau is a user-defined parameter
-            # set up gaussian process using celerite
-            # we use a Matern-3/2 kernel for its flexibility and non-periodicity
-            kernel = celerite.terms.Matern32Term(np.log(amp), np.log(tau))
-            gp = celerite.GP(kernel)
-            gp.compute(M(self.time), M(rawflux_err))
-
-            # compute the coefficients C on the basis vectors;
-            # the PLD design matrix will be dotted with C to solve for the noise model.
-            A = np.dot(MX.T, gp.apply_inverse(MX))
-            B = np.dot(MX.T, gp.apply_inverse(M(rawflux)[:, None])[:, 0])
+            cadence_mask = np.ones(len(self.raw_lc.time), dtype=bool)
+
+        # Theano raises an `AsTensorError` ("Cannot convert to TensorType")
+        # if the floats are not in 64-bit format, so we convert them here:
+        lc_time = np.asarray(self.raw_lc.time, np.float64)
+        lc_flux = np.asarray(self.raw_lc.flux, np.float64)
+        lc_flux_err = np.asarray(self.raw_lc.flux_err, np.float64)
+
+        # Covariance matrix diagonal
+        diag = lc_flux_err**2
+
+        # The cadence mask is applied by inflating the uncertainties in the covariance matrix;
+        # this is because celerite will run much faster if it is able to predict
+        # data for cadences that have been fed into the model.
+        diag[~cadence_mask] += 1e12
+
+        with pm.Model() as model:
+            # Create a Gaussian Process to model the long-term stellar variability
+            # log(sigma) is the amplitude of variability, estimated from the raw flux scatter
+            logsigma = pm.Normal("logsigma", mu=np.log(np.std(lc_flux)), sd=4)
+            # log(rho) is the timescale of variability with a user-defined prior
+            logrho = pm.Normal("logrho", mu=np.log(gp_timescale_prior), sd=4)
+            # log(s2) is a jitter term to compensate for underestimated flux errors
+            logs2 = pm.Normal("logs2", mu=np.log(np.var(lc_flux)), sd=4)
+            kernel = xo.gp.terms.Matern32Term(log_sigma=logsigma, log_rho=logrho)
+
```

Comment:
Add the mean as a parameter, otherwise you will end up with lots of funny negatives as discussed in office.

It will go super negative.